### PR TITLE
chore(docs): Update proving backend related docs

### PR DESCRIPTION
**Pull request by Savio-Sou**
# Description

## Summary\*

Update docs related to the installation and use of proving backends following https://github.com/noir-lang/noir/issues/4960.

Focusing mainly on improving:
1. Noir's proving backend agnosticity
2. Formatting/readability

Best to merge this PR in before stable releasing v0.31.0 and v0.32.0.

## Documentation\*

Check one:
- [ ] No documentation needed.
- [x] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by Savio-Sou** [comment]
@jzaki / @signorecello / others grateful if could help sanity check if the changes make sense to read.

File: docs/docs/getting_started/hello_noir/index.md [hunk 1]
```diff
@@ -17,41 +17,41 @@ sidebar_position: 1
...
 
-> **Note:** `hello_world` can be any arbitrary project name, we are simply using `hello_world` for
-> demonstration.
->
-> In production, the common practice is to name the project folder as `circuits` for better
-> identifiability when sitting alongside other folders in the codebase (e.g. `contracts`, `scripts`,
-> `test`).
+`hello_world` can be any arbitrary project name, we are simply using `hello_world` for demonstration.
+
+In production, the common practice is to name the project folder as `circuits` for better identifiability when sitting alongside other folders in the codebase (e.g. `contracts`, `scripts`, `test`).
```
Comment (jzaki) [inline docs/docs/getting_started/hello_noir/index.md:48]:
```suggestion
In production, it is common practice to name the project folder, `circuits`, for clarity amongst other folders in the codebase (like: `contracts`, `scripts`, and `test`).
```

File: docs/docs/getting_started/hello_noir/index.md [hunk 2]
```diff
@@ -111,34 +111,41 @@ nargo execute witness-name
...
-## Prove Our Noir Program
+The command also automatically compiles your Noir program if it was not already / was edited, which you may notice the compiled artifacts being written to the file `./target/hello_world.json`.
+
+## Proving Backend
 
-:::info
+Proving backends provide the ability to generate and verify proofs of executing Noir programs, following Noir's tooling that compiles and executes the programs. Read the [proving backend installation](../backend/index.md) section to learn more about proving backends and how to install them.
 
-Nargo no longer handles communicating with backends in order to generate proofs. In order to prove/verify your Noir programs, you'll need an installation of [bb](../barretenberg/index.md).
+Barretenberg is used as an example here to demonstrate how proving and verifying could be implemented and used. Read the [`bb` installation](../backend/index.md#example-installing-bb) section for how to install Barretenberg's CLI tool; refer to [`bb`'s documentations](https://github.com/AztecProtocol/barretenberg/blob/master/cpp/src/barretenberg/bb/readme.md) for full details about the tool.
```
Comment (jzaki) [inline docs/docs/getting_started/hello_noir/index.md:120]:
```suggestion
Barretenberg is used as an example here to demonstrate how proving and verifying could be implemented and used. Read the [`bb` installation](../backend/index.md#example-installing-bb) section for how to install Barretenberg's CLI tool; refer to [`bb`'s documentation](https://github.com/AztecProtocol/barretenberg/blob/master/cpp/src/barretenberg/bb/readme.md) for full details about the tool.
```

File: docs/docs/getting_started/hello_noir/index.md [hunk 3]
```diff
@@ -111,34 +111,41 @@ nargo execute witness-name
...
-A new file called `proof` will be generated in your project directory, containing the generated proof for your program.
+The proof generated will then be written to the file `./target/proof`.
 
-## Verify Our Noir Program
+### 6. Verify the execution proof
 
 Once a proof is generated, we can verify correct execution of our Noir program by verifying the proof file.
 
-Verify your proof by running:
+Using Barretenberg as an example, generate a verification key for your Noir program running:
```
Comment (jzaki) [inline docs/docs/getting_started/hello_noir/index.md:136]:
```suggestion
Using Barretenberg as an example, compute/output the verification key for the Noir program by running:
```
As I understand it, the proving key would already be generated, and the verification key is calculated from it.

File: docs/docs/getting_started/installation/index.md [hunk 4]
```diff
@@ -19,11 +19,9 @@ keywords: [
 pagination_next: getting_started/hello_noir/index
 ---
 
-`nargo` is the one-stop-shop for almost everything related with Noir. The name comes from our love for Rust and its package manager `cargo`.
+`nargo` is a tool for working with Noir programs on the CLI, providing you the ability to start new projects, compile, execute and test Noir programs from the terminal.
```
Comment (jzaki) [inline docs/docs/getting_started/installation/index.md:22]:
```suggestion
`nargo` is a tool for working with Noir programs on the CLI, providing you with the ability to start new projects, compile, execute and test Noir programs from the terminal.
```

File: docs/versioned_docs/version-v0.32.0/getting_started/backend/index.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
+---
```
Comment (jzaki) [inline docs/versioned_docs/version-v0.32.0/getting_started/backend/index.md:1]: Same comments as per `docs/docs/getting_started/backend/index.md` above.

File: docs/versioned_docs/version-v0.31.0/getting_started/backend/index.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
+---
```
Comment (jzaki) [inline docs/versioned_docs/version-v0.31.0/getting_started/backend/index.md:1]: Same comments as per `docs/docs/getting_started/backend/index.md` above.

File: docs/docs/getting_started/backend/index.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+For the latest information on tooling provided by each proving backend, installation instructions, Noir version compatibility... you may refer to the proving backends' own documentations.
+
+You can find the full list of proving backends compatible with Noir in [Awesome Noir](https://github.com/noir-lang/awesome-noir/?tab=readme-ov-file#proving-backends).
+
+## Example: Installing `bb`
+
+`bb` is the CLI tool provided by the [Barretenberg proving backend](https://github.com/AztecProtocol/barretenberg) developed by Aztec Labs.
+
+As an example of how a proving backend could be installed, you can install `bb` running the commands below in a terminal.
```
[the remark this answers is not in the document]
Comment (Savio-Sou) [reply docs/docs/getting_started/backend/index.md:29]: Page would look super empty 😂 but agree best to not maintain the same thing at two places

File: docs/docs/getting_started/backend/index.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    ```bash
+    source ~/.zshrc
+    ```
+
+    Linux:
+    ```bash
+    source ~/.bashrc
+    ```
+
+3. Install the version of `bb` compatible with your Noir version:
```
[the remark this answers is not in the document]
Comment (Savio-Sou) [reply docs/docs/getting_started/backend/index.md:49]: > And at least below will be in sync with what is compatible at the time of writing.

That's my plan until https://github.com/AztecProtocol/aztec-packages/issues/7511 is addressed, then we can simply link to it for readers' reference.

[the remark this answers is not in the document]
Comment (Savio-Sou) [reply docs/docs/getting_started/backend/index.md:49]: > Does this repo have any macros like #include_aztec_version? Could have an #include_bb_version, at least while bb is the default.

I'm not quite familiar with that macro 🤔 could you elaborate on what does it do?

**Review by jzaki — APPROVED** [review]

**Pull request merged**